processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -94,20 +94,44 @@ public AggregateCombiner makeNullableAggregateCombiner()
   }
 
   /**
-   * Returns an AggregatorFactory that can be used to combine the output of aggregators from this factory.  This
-   * generally amounts to simply creating a new factory that is the same as the current except with its input
-   * column renamed to the same as the output column.
+   * Returns an AggregatorFactory that can be used to combine the output of aggregators from this factory. It is used
+   * when we know we have some values that were produced with this aggregator factory, and want to do some additional
+   * combining of them. This happens, for example, when merging query results from two different segments, or two
+   * different servers.
+   *
+   * The combining factory is often computed by simply creating a new factory that is the same as the current, except
```

Comment:
Doesn't sound quite right to me. E. g. the majority of complex aggregators have different combining factories.

Reply:
I changed the wording a bit to refer to this being true for "simple aggregators" rather than "often".